mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -108,11 +109,20 @@ enum TxStatus {
 }
 
 // Structure used in specifying the list of outputs when generating a transaction.
+// Here the token id is implied from context, and matches the fee token id.
 message Outlay {
     uint64 value = 1;
     external.PublicAddress receiver = 2;
 }
 
+// Structure used in specifying the list of outputs in a transaction.
+// Here the token id is explicit.
+message OutlayV2 {
```

Comment:
You can't tell if its missing or of it is intentionally set to 0 to represent MOB

Reply:
Oh I didn't realize that they didn't serialize default values. That's kind of surprising.

Reply:
I think a work around that might be interesting is to use a oneof for the token id field, so it defaults to a pseudo null value, instead of to 0.

Reply:
I think that's a more complicated approach than what we did here

Reply:
I don't want all the clients that use this to have to deal with enums and "pseudo-null values" and one ofs and so on.

I like that with OutlayV2, you can be 100% sure that the token id is present in a meaningful way. And with Outlay v1, you can be 100% sure that the token id is not present and has to be inferred from context. And we can use the appropriate one in APIs as appropriate.

Reply:
I guess that's fair. The only thing I'm really trying to avoid is having to distinguish outlay and outlayv2 scenarios and make sure that the default behaviour is consistent, but at the end of the day, having to check a null value is similar.

Reply:
> I think a work around that might be interesting is to use a oneof for the token id field, so it defaults to a pseudo null value, instead of to 0.

thinking about this more, I don't think this would work either.

Protobuf serializers are allowed to omit defaulted values, even if the user explicitly set token id to 0 (MOB).

If we change what the deserializer thinks that means, from 0 to "null", that's a breaking change. So I don't think the oneof idea really helps anything -- the default value, when the user omitted this field, has to be zero.